### PR TITLE
Implementation of ShoppingCart

### PR DESCRIPTION
**Pull request by holsoni**

### CODE REVIEW

File: src/main/java/com/example/bookstore/dto/shoppingcart/CartItemDto.java [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package com.example.bookstore.dto.shoppingcart;
+
+import lombok.Data;
+
+@Data
+public class CartItemDto {
```
Comment (andrii-hoienko) [inline src/main/java/com/example/bookstore/dto/shoppingcart/CartItemDto.java:6]: is it possible to add some validations to DTOs or if for example, the quantity will be -10 it is ok?

File: src/main/java/com/example/bookstore/service/shoppingcart/impl/ShoppingCartServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+                    newCart.setUser(currentUser);
+                    return shoppingCartRepository.save(newCart);
+                });
+        return shoppingCartMapper.toDto(shoppingCart);
+    }
+
+    @Override
+    public CartItemDto addToCart(AddCartItemRequestDto requestDto, Authentication authentication) {
+        User currentUser = (User) authentication.getPrincipal();
+        ShoppingCart shoppingCart = shoppingCartRepository.findByUserId(currentUser.getId())
```
Comment (andrii-hoienko) [inline src/main/java/com/example/bookstore/service/shoppingcart/impl/ShoppingCartServiceImpl.java]: Triple code duplication (addToCart, getShoppingCart, create)

File: src/main/resources/db/changelog/db.changelog-master.yaml [hunk 3]
```diff
@@ -17,3 +17,9 @@ databaseChangeLog:
       file: db/changelog/changes/08-create-book_category-table.yaml
   - include:
       file: db/changelog/changes/09-update-category-table.yaml
+  - include:
+      file: db/changelog/changes/10-create-shopping_cart-table.yaml
+  - include:
+      file: db/changelog/changes/11-create-cart_item-table.yaml
+  - include:
+      file: db/changelog/changes/12-create-shopping_car_cart_item-table.yaml
```
Comment (andrii-hoienko) [inline src/main/resources/db/changelog/db.changelog-master.yaml]: add empty line please

File: src/main/resources/db/changelog/changes/12-create-shopping_car_cart_item-table.yaml [hunk 4]
```diff
@@ -0,0 +1,24 @@
+databaseChangeLog:
+- changeSet:
+    id: create-shopping_cart_cart_item-table
+    author: holsoni
```
Comment (andrii-hoienko) [inline src/main/resources/db/changelog/changes/12-create-shopping_car_cart_item-table.yaml]: It is difficult to identify author of change set, please set your name and surname in author section

File: src/main/java/com/example/bookstore/service/user/impl/UserServiceImpl.java [hunk 5]
```diff
@@ -34,7 +33,8 @@ public UserResponseDto register(UserRegistrationRequestDto requestDto)
         Role role = roleRepository.getByName(RoleName.ROLE_USER);
         User user = userMapper.toModel(requestDto);
         user.setRoles(Set.of(role));
-        user.setPassword(new BCryptPasswordEncoder().encode(requestDto.getPassword()));
+        user.setPassword(passwordEncoder.encode(requestDto.getPassword()));
         return userMapper.toDto(userRepository.save(user));
     }
+
```
Comment (andrii-hoienko) [inline src/main/java/com/example/bookstore/service/user/impl/UserServiceImpl.java]:
```suggestion
```

File: src/main/java/com/example/bookstore/service/shoppingcart/impl/ShoppingCartServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+                    ShoppingCart newCart = new ShoppingCart();
+                    newCart.setUser(currentUser);
+                    return shoppingCartRepository.save(newCart);
+                });
+        Book book = bookRepository.findById(requestDto.getBookId()).orElseThrow(
+                () -> new EntityNotFoundException("Sorry! There is no book with id "
+                        + requestDto.getBookId()));
+
+        CartItem cartItem = cartItemMapper.toEntity(requestDto);
+        cartItem.setBook(book);
```
Comment (andrii-hoienko) [inline src/main/java/com/example/bookstore/service/shoppingcart/impl/ShoppingCartServiceImpl.java]: Is it possible to move mapping logic to mappers?

**Review by andrii-hoienko — CHANGES_REQUESTED** [review]
Nice, but some improvements required

File: src/main/java/com/example/bookstore/dto/shoppingcart/AddCartItemRequestDto.java [hunk 7]
```diff
@@ -0,0 +1,11 @@
+package com.example.bookstore.dto.shoppingcart;
+
+import jakarta.validation.constraints.Min;
+import lombok.Data;
+
+@Data
+public class AddCartItemRequestDto {
+    private Long bookId;
```
Comment (vovasalyha) [inline src/main/java/com/example/bookstore/dto/shoppingcart/AddCartItemRequestDto.java:10]: can bookId be null?

**Review by vovasalyha — APPROVED** [review]

**Pull request merged**